core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2153,6 +2153,9 @@ private[spark] object Utils extends Logging {
       conf.getInt("spark.executor.instances", 0) == 0
   }
 
+  def withResource[R <: Closeable, T](resource: R, f: () => T): T = {
```

Comment:
What do you think about naming this `tryWithResource` to evoke Java's try-with-resources statement?